In [7]:
import pandas as pd

penetracion = pd.read_csv("Internet_Penetracion.csv", decimal=',')

display(penetracion.head(10))
penetracion.info()


Año  Trimestre        Provincia  Accesos por cada 100 hogares
0  2022          2     Buenos Aires                         77.58
1  2022          2  Capital Federal                        121.31
2  2022          2        Catamarca                         62.40
3  2022          2            Chaco                         43.79
4  2022          2           Chubut                         83.19
5  2022          2          Córdoba                         87.48
6  2022          2       Corrientes                         48.35
7  2022          2       Entre Ríos                         63.90
8  2022          2          Formosa                         43.42
9  2022          2            Jujuy                         57.32

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 816 entries, 0 to 815
Data columns (total 4 columns):
 #   Column                        Non-Null Count  Dtype  
---  ------                        --------------  -----  
 0   Año                           816 non-null    int64  
 1   Trimestre                     816 non-null    int64  
 2   Provincia                     816 non-null    object 
 3   Accesos por cada 100 hogares  816 non-null    float64
dtypes: float64(1), int64(2), object(1)
memory usage: 25.6+ KB


In [4]:
penetracion['Accesos por cada 100 hogares'].max()

124.06

En este dataset tenemos el histórico trimestral de la cantidad de accessos por cada 100 hogares, desglozado por provincia. Tenemos 816 registros desde 2014 hasta 2021. Uno de los objetivos de este proyecto es evaluar el aumento o disminución de la variación porcentual trimestral del servicio de internet, cada 100 hogares por provincia y este dataset tiene toda la informacion que necesitamos para cumplir este objetivo.

In [5]:

conectividad =pd.read_csv("ConectividadalserviciodeInternet_2776151672760470518.csv")

conectividad.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4312 entries, 0 to 4311
Data columns (total 16 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Provincia      4312 non-null   object 
 1   Partido        4312 non-null   object 
 2   Localidad      4312 non-null   object 
 3   Poblacion      4312 non-null   int64  
 4   ADSL           4312 non-null   object 
 5   CABLEMODEM     4312 non-null   object 
 6   DIALUP         4312 non-null   object 
 7   FIBRAOPTICA    4312 non-null   object 
 8   SATELITAL      4312 non-null   object 
 9   WIRELESS       4312 non-null   object 
 10  TELEFONIAFIJA  4312 non-null   object 
 11  3G             4312 non-null   object 
 12  4G             4312 non-null   object 
 13  link           3836 non-null   float64
 14  Latitud        4312 non-null   float64
 15  Longitud       4312 non-null   float64
dtypes: float64(3), int64(1), object(12)
memory usage: 539.1+ KB


En este data set tenemos el desgloce de conexiones a internet disponibles por partido y localidad, desglosado en el tipo de conexion. Tambien tenemos la poblacion de cada localidad, de aqui podemos extraer informacion como que porcentage de la poblacion tiene acceso a internet. Tenemos 4312 registros, de todas las localidades del pais. 

In [5]:
conectividad[(conectividad['ADSL'] == '--') & (conectividad['CABLEMODEM'] == '--') & (conectividad['DIALUP'] == '--' )& (conectividad['FIBRAOPTICA'] == '--') & 
    (conectividad['SATELITAL'] == '--') & (conectividad['WIRELESS'] == '--') & (conectividad['3G'] == '--') & (conectividad['4G'] == '--')]

Provincia                 Partido           Localidad  Poblacion  \
1     BUENOS AIRES              25 de Mayo     Agustín Mosconi        280   
8     BUENOS AIRES              25 de Mayo         San Enrique        279   
13    BUENOS AIRES          Carlos Tejedor              Curarú        422   
14    BUENOS AIRES          Carlos Tejedor              Timote        526   
28    BUENOS AIRES       General La Madrid           La Colina        663   
...            ...                     ...                 ...        ...   
4285       FORMOSA                 Bermejo         Guadalcazar        146   
4298         JUJUY  Doctor Manuel Belgrano             Ocloyas         70   
4299         JUJUY  Doctor Manuel Belgrano  San Pablo De Reyes          0   
4303         JUJUY               El Carmen   Barrio El Milagro       1321   
4304         JUJUY               El Carmen     Barrio La Unión        700   

     ADSL CABLEMODEM DIALUP FIBRAOPTICA SATELITAL WIRELESS TELEFONIAFIJA  3G  \
1      --         --     --          --        --       --            SI  --   
8      --         --     --          --        --       --            SI  --   
13     --         --     --          --        --       --            SI  --   
14     --         --     --          --        --       --            SI  --   
28     --         --     --          --        --       --            SI  --   
...   ...        ...    ...         ...       ...      ...           ...  ..   
4285   --         --     --          --        --       --            --  --   
4298   --         --     --          --        --       --            --  --   
4299   --         --     --          --        --       --            --  --   
4303   --         --     --          --        --       --            --  --   
4304   --         --     --          --        --       --            --  --   

      4G        link    Latitud   Longitud  
1     --   6854010.0 -35.739217 -60.559063  
8     --   6854080.0 -35.778503 -60.354645  
13    --   6154030.0 -35.640350 -62.192437  
14    --   6154040.0 -35.347762 -62.224653  
28    --   6322020.0 -37.360769 -61.534849  
...   ..         ...        ...        ...  
4285  --  34007005.0 -23.682394 -61.161109  
4298  --  38021050.0 -23.909654 -65.195507  
4299  --         NaN -24.143659 -65.482920  
4303  --  38014020.0 -24.461236 -65.119054  
4304  --  38014030.0 -24.403542 -65.068590  

[904 rows x 16 columns]

Aqui podemos evidenciar que efectivamente hay 904 localidades que no tienen acceso a internet

In [12]:
conectividad[(conectividad['Poblacion'] == 0)]

Provincia                 Partido                Localidad  \
35    BUENOS AIRES       General Las Heras                   Lozano   
38    BUENOS AIRES         General Lavalle  Chacras de San Clemente   
42    BUENOS AIRES                La Plata           La Providencia   
43    BUENOS AIRES                La Plata           Lisandro Olmos   
45    BUENOS AIRES                La Plata               Los Hornos   
...            ...                     ...                      ...   
4238       CORDOBA              San Javier              Los Romeros   
4253  BUENOS AIRES  General Juan Madariaga           Barrio Kennedy   
4268       FORMOSA                 Formosa           Villa Trinidad   
4299         JUJUY  Doctor Manuel Belgrano       San Pablo De Reyes   
4306         JUJUY               El Carmen                  Fleming   

      Poblacion ADSL CABLEMODEM DIALUP FIBRAOPTICA SATELITAL WIRELESS  \
35            0   --         --     --          --        SI       --   
38            0   --         --     --          --        --       SI   
42            0   --         --     --          --        --       --   
43            0   --         SI     --          --        --       --   
45            0   --         SI     --          SI        --       SI   
...         ...  ...        ...    ...         ...       ...      ...   
4238          0   --         --     --          --        --       SI   
4253          0   --         --     --          --        --       --   
4268          0   --         --     --          --        --       --   
4299          0   --         --     --          --        --       --   
4306          0   --         --     --          --        --       --   

     TELEFONIAFIJA  3G  4G       link    Latitud   Longitud  
35              --  --  --  6329020.0 -34.854280 -59.053345  
38              --  --  --  6336020.0 -36.408085 -56.943355  
42              --  --  --  6441030.0 -35.023248 -58.230703  
43              --  SI  SI  6441030.0 -34.997609 -58.043377  
45              --  SI  SI  6441030.0 -34.973716 -57.985054  
...            ...  ..  ..        ...        ...        ...  
4238            --  --  --        NaN -32.137967 -65.223236  
4253            --  SI  SI        NaN -37.006222 -57.141726  
4268            --  --  --        NaN -26.182822 -58.173393  
4299            --  --  --        NaN -24.143659 -65.482920  
4306            --  SI  SI        NaN -24.587376 -64.924058  

[546 rows x 16 columns]

In [11]:
conectividad[(conectividad['Poblacion'] == 0) & (conectividad['ADSL'] == '--') & (conectividad['CABLEMODEM'] == '--') & (conectividad['DIALUP'] == '--' )& (conectividad['FIBRAOPTICA'] == '--') & 
    (conectividad['SATELITAL'] == '--') & (conectividad['WIRELESS'] == '--') & (conectividad['3G'] == '--') & (conectividad['4G'] == '--')]

Provincia                     Partido                Localidad  \
42    BUENOS AIRES                    La Plata           La Providencia   
47    BUENOS AIRES                    La Plata                   Poblet   
127       SAN LUIS  Libertador Grl. San Martín         Planta De Sandia   
153        CORDOBA                 Santa María  Barrio Villa del Parque   
186   BUENOS AIRES               Adolfo Alsina                   Thames   
...            ...                         ...                      ...   
4218       MENDOZA               Luján de Cuyo         Villa El Refugio   
4226       CORDOBA                  San Javier              El Pueblito   
4227       CORDOBA                  San Javier                 El Valle   
4268       FORMOSA                     Formosa           Villa Trinidad   
4299         JUJUY      Doctor Manuel Belgrano       San Pablo De Reyes   

      Poblacion ADSL CABLEMODEM DIALUP FIBRAOPTICA SATELITAL WIRELESS  \
42            0   --         --     --          --        --       --   
47            0   --         --     --          --        --       --   
127           0   --         --     --          --        --       --   
153           0   --         --     --          --        --       --   
186           0   --         --     --          --        --       --   
...         ...  ...        ...    ...         ...       ...      ...   
4218          0   --         --     --          --        --       --   
4226          0   --         --     --          --        --       --   
4227          0   --         --     --          --        --       --   
4268          0   --         --     --          --        --       --   
4299          0   --         --     --          --        --       --   

     TELEFONIAFIJA  3G  4G       link    Latitud   Longitud  
42              --  --  --  6441030.0 -35.023248 -58.230703  
47              --  --  --  6441030.0 -35.060458 -57.954743  
127             --  --  --        NaN -32.767993 -65.548852  
153             --  --  --        NaN -31.777219 -64.532078  
186             --  --  --  6007010.0 -36.929166 -63.295727  
...            ...  ..  ..        ...        ...        ...  
4218            --  --  --        NaN -32.997530 -69.317680  
4226            --  --  --        NaN -31.951097 -65.072151  
4227            --  --  --        NaN -31.951097 -65.072151  
4268            --  --  --        NaN -26.182822 -58.173393  
4299            --  --  --        NaN -24.143659 -65.482920  

[267 rows x 16 columns]

Aqui podemos evidenciar que tenemos 267 filas que tienen poblacion 0 y que no reportan acceso a internet. Se debe tener en cuenta en el analisis esta información faltante. 

In [11]:
velocidad = pd.read_csv("historico_velocidad_internet.csv")

display(velocidad.head())
velocidad.info()

Año  Trimestre        Provincia  Mbps (Media de bajada)
0  2022          2     Buenos Aires                      66
1  2022          2  Capital Federal                      94
2  2022          2        Catamarca                      57
3  2022          2            Chaco                      51
4  2022          2           Chubut                      15

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 816 entries, 0 to 815
Data columns (total 4 columns):
 #   Column                  Non-Null Count  Dtype 
---  ------                  --------------  ----- 
 0   Año                     816 non-null    int64 
 1   Trimestre               816 non-null    int64 
 2   Provincia               816 non-null    object
 3   Mbps (Media de bajada)  816 non-null    int64 
dtypes: int64(3), object(1)
memory usage: 25.6+ KB


En este dataset tenemos el historico trimestral de la velocidad promedio de bajada de internet, desglozada por provincia. Con este dataset podemos encontrar la variación con el tiempo y tambien usar la información de la media para descubrir otros datos. 

In [12]:
ingresos = pd.read_csv("Internet_Ingresos.csv")

display(ingresos.head())
ingresos.info()

Año  Trimestre Ingresos (miles de pesos)        Periodo
0  2022          2                60.942.843   Abr-Jun 2022
1  2022          1                56.195.811   Ene-Mar 2022
2  2021          4                45.467.887   Oct-Dic 2021
3  2021          3                42.999.944  Jul-Sept 2021
4  2021          2                38.239.667   Abr-Jun 2021

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 34 entries, 0 to 33
Data columns (total 4 columns):
 #   Column                     Non-Null Count  Dtype 
---  ------                     --------------  ----- 
 0   Año                        34 non-null     int64 
 1   Trimestre                  34 non-null     int64 
 2   Ingresos (miles de pesos)  34 non-null     object
 3   Periodo                    34 non-null     object
dtypes: int64(2), object(2)
memory usage: 1.2+ KB


En este dataset tenemos el histórico trimestral de los ingresos por la prestación del servicio de internet fijo. Con esta información podemos ver el cambio histórico de ingresos y teniendo en cuenta la cantidad de accesos y el precio, podemos conocer el precio promedio de un Mbps de forma histórica. 


In [6]:
accessos = pd.read_csv("Internet_Accesos-por-velocidad.csv")

display(accessos.head())
accessos.info()

Año  Trimestre        Provincia  HASTA 512 kbps  + 512 Kbps - 1 Mbps  \
0  2022          2     Buenos Aires          30.639               28.323   
1  2022          2  Capital Federal         517.000                5.980   
2  2022          2        Catamarca          71.000              413.000   
3  2022          2            Chaco         461.000                1.047   
4  2022          2           Chubut         109.000                1.444   

  + 1 Mbps - 6 Mbps  + 6 Mbps - 10 Mbps  + 10 Mbps - 20 Mbps  \
0           295.238             307.554              273.954   
1            36.036              72.434               55.372   
2             3.678               6.155                7.165   
3            17.603              20.109                8.391   
4            45.804              31.217               34.290   

   + 20 Mbps - 30 Mbps  + 30 Mbps    OTROS      Total  
0              129.869  3.535.757   65.849  4.667.183  
1               31.571  1.229.254  105.607  1.536.771  
2                3.393     43.926    2.207     67.008  
3               15.974     76.271    3.708    143.564  
4               14.526     15.394   20.021    162.805

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 816 entries, 0 to 815
Data columns (total 12 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   Año                  816 non-null    int64  
 1   Trimestre            816 non-null    int64  
 2   Provincia            816 non-null    object 
 3   HASTA 512 kbps       816 non-null    float64
 4   + 512 Kbps - 1 Mbps  816 non-null    float64
 5   + 1 Mbps - 6 Mbps    816 non-null    object 
 6   + 6 Mbps - 10 Mbps   816 non-null    float64
 7   + 10 Mbps - 20 Mbps  816 non-null    float64
 8   + 20 Mbps - 30 Mbps  816 non-null    float64
 9   + 30 Mbps            816 non-null    object 
 10  OTROS                810 non-null    float64
 11  Total                816 non-null    object 
dtypes: float64(6), int64(2), object(4)
memory usage: 76.6+ KB


De este dataset podemos conocer la cantidad de conexiones por velocidad de conexion en cada region de forma historica. De esta información podemos usar el total de conexiones para determinar, junto con el promedio de descarga, la capacidad total instalada en Mbps. Con esto y el ingreso, podemos determinar el costo por Mbps historico. 

Tambien podemos conocer la moda de la velocidad de conexion, que en este caso es relevante pues el promedio no nos da toda la información sobre los tipos de conexion instaladas.